wgengine/filter/filter.go

```diff
@@ -206,12 +208,14 @@ func matchesFamily(ms matches, keep func(netip.Addr) bool) matches {
 				retm.Srcs = append(retm.Srcs, src)
 			}
 		}
+
 		for _, dst := range m.Dsts {
 			if keep(dst.Net.Addr()) {
 				retm.Dsts = append(retm.Dsts, dst)
 			}
 		}
 		if len(retm.Srcs) > 0 && len(retm.Dsts) > 0 {
+			retm.SrcsContains = tsaddr.NewContainsIPFunc(views.SliceOf(retm.Srcs))
```

Comment:
now we always just populate this func, which does whatever's fastest depending on Srcs (per #12487)